src/wallet/wallet.cpp

```diff
@@ -389,15 +388,10 @@ void CWallet::UpgradeKeyMetadata()
             // Write meta to wallet
             CPubKey pubkey;
             if (GetPubKey(meta_pair.first, pubkey)) {
-                batch->WriteKeyMetadata(meta, pubkey, true);
-                if (++cnt % 1000 == 0) {
-                    // avoid creating overlarge in-memory batches in case the wallet contains large amounts of keys
-                    batch.reset(new WalletBatch(*database));
-                }
+                batch.WriteKeyMetadata(meta, pubkey, true);
             }
         }
     }
-    batch.reset(); //write before setting the flag
```

Comment:
in commit ebb0ea85539b3e2f515fca89231c20ac86fb52d5:

Why is this no longer needed?

Reply:
When batch goes out of scope, it will write. This was just a belt and suspenders.

Reply:
The comment suggests that in this case -- unlike the batch write in importmulti -- there's a specific reason to force it to write before the end of the scope.

Reply:
reverted